### PR TITLE
Skip unit tests that fail on MI210 runners

### PR DESCRIPTION
**Pull request by jithunnair-amd**
Taken from https://github.com/pytorch/pytorch/pull/105980

Skip following for ROCm:
* test_nn.py::TestConvolutionNNDeviceTypeCUDA::test_conv_transposed_large_cuda (until ROCm6.0)
* test_ops.py::TestCommonCUDA::test_complex_half_reference_testing_nn_functional_conv_transpose3d_cuda_complex32
* test_matmul_cuda.py::TestMatmulCudaCUDA::test_cublas_baddbmm_large_input_*

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
@malfet Pretty straightforward skips for ROCm, to get green CI on MI210 runners. Please approve.

**Review by malfet — APPROVED** [review]

File: test/nn/test_convolution.py [hunk 1]
```diff
@@ -1927,6 +1927,7 @@ def test_conv_noncontig_weights_and_bias(self, device):
 
     @onlyCUDA
     @largeTensorTest('12GB')
+    @skipIfRocmVersionLessThan((6, 0))
```
Comment (malfet) [inline test/nn/test_convolution.py:1930]: Just to be clear, this feels like it skips it if ROCm version is older than X rather then if runner type is older, isn't it? So title is a bit misleading

Comment (jithunnair-amd) [reply test/nn/test_convolution.py:1930]: I agree it's a bit confusing, but the failure is a MI210-specific failure which will be fixed in ROCm6.0, hence the `skipIfRocmVersionLessThan` decorator

**Pull request closed without merging**